File: java/examples-java-junit-maven/src/test/java/examples/YahtzeeSteps.java
package examples;

import com.oselvar.var.Registrar;
import com.oselvar.var.State;
import com.oselvar.var.StateBinder;
import com.oselvar.var.StepDefinitions;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

public final class YahtzeeSteps implements StepDefinitions {

    record Ctx() implements State {}

    @Override
    public void defineSteps(Registrar registrar) {
        StateBinder<Ctx> s = registrar.defineState(Ctx::new);

        // Header-bound table: the paragraph names every header cell (dice,
        // category, score), so this sensor runs once per row with the row as a
        // Map keyed by header. Returning Map.of("score", …) checks that column;
        // the other columns are inputs.
        s.sensor("Examples of dice, category and score", (Ctx ctx, Map<String, String> row) -> {
            List<Integer> dice = Arrays.stream(row.get("dice").split(","))
                    .map(d -> Integer.parseInt(d.trim()))
                    .toList();
            return Map.of("score", Yahtzee.score(dice, row.get("category")));
        });
    }
}

File: examples/python-pytest/steps/yahtzee.steps.py
from var import define_state
from yahtzee_example import score

stimulus, sensor = define_state()


# Header-bound table: the paragraph names every header cell (dice, category,
# score), so this sensor runs once per row with the row as a dict keyed by
# header. Returning {"score": …} checks that column; the other columns are
# inputs.
@sensor("Examples of dice, category and score")
def _(state, row):
    dice = [int(d.strip()) for d in row["dice"].split(",")]
    return {"score": score(dice, row["category"])}
